assessment/collectors/app_migration_analysis.go

```diff
@@ -47,17 +48,36 @@ type MigrationSummarizer struct {
 	projectPath                   string
 	dependencyGraph               map[string]map[string]struct{}
 	fileDependencyAnalysisDataMap map[string]FileDependencyAnalysisData
+	RunParallel                   task.RunParallelTasksInterface[*AnalyzeFileInput, *AnalyzeFileResponse]
```

Comment:
this does not need to be in this struct - it can just directly be called as a utils method where needed.

Reply:
Done.